sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/DebugOnlyCodeHelper.cs

```diff
@@ -0,0 +1,83 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Helpers
+{
+    internal static class DebugOnlyCodeHelper
+    {
+        // Looking for an exact case-sensitive match
+        public static bool IsDebugString(string text) =>
```

Comment:
Moved from the _ConsoleLogging_ rule into a helper class.

I've deleted some of the code as I realised I'd made a mistake in the original version. The MSDN documentation for _ConditionalAttribute_ says it can be applied to methods or classes. However, the small-print in the documentation says it does nothing at class level unless it's on an attribute class, so I removed the walk up the class hierarchy to check for _[Conditional(DEBUG)]_ at class level.